tests/core/utilities/test_valid_transaction_params.py

```diff
@@ -38,20 +38,49 @@ def test_assert_valid_transaction_params_invalid_param():
 
 
 def test_extract_valid_transaction_params():
+    ''' baseline test for clean param dict '''
     input = {
         'from': '0x0',
         'to': '0x0',
         'gas': 21000,
         'gasPrice': 5000000,
-        'value': 1,
         'data': '0x0',
+        'value': 1,
         'nonce': 2,
         'chainId': 1,
     }
     valid_transaction_params = extract_valid_transaction_params(input)
     assert valid_transaction_params == input
 
 
+test_data = [({'data': '0x0', 'input': '0x0'}, {'data': '0x0'}), ]
+@pytest.mark.parametrize("transaction_params, expected", test_data)  # noqa E302
+def test_extract_valid_transaction_params_2(transaction_params, expected):
+    valid_transaction_params = extract_valid_transaction_params(transaction_params)
+    assert valid_transaction_params == expected
+
+
+test_data = [({'input': '0x0'}, {'data': '0x0'}), ]
+@pytest.mark.parametrize("transaction_params, expected", test_data)  # noqa E302
+def test_extract_valid_transaction_params_3(transaction_params, expected):
+    valid_transaction_params = extract_valid_transaction_params(transaction_params)
+    assert valid_transaction_params == expected
+
+
+test_data = [({}, {}), ]
+@pytest.mark.parametrize("transaction_params, expected", test_data)  # noqa E302
+def test_extract_valid_transaction_params_4(transaction_params, expected):
+    valid_transaction_params = extract_valid_transaction_params(transaction_params)
+    assert valid_transaction_params == expected
+
+
+
+@pytest.mark.xfail(raises=AttributeError, strict=True)  # noqa E302
+def test_extract_valid_transaction_params_5():
+    transaction_params = {'data': '0x0', 'input': '0x1'}
+    extract_valid_transaction_params(transaction_params)
```

Comment:
Sometimes we combine the exceptional cases into the standard cases, but I'm personally fine with having this be a separate test, which keeps the test body of both tests simple.

However, we have been using `xfails` exclusively for known bugs. Testing that we intentionally raise exceptions should be done with `with pytest.raises(AttributeError):` instead. Bonus points if you use the `match` keyword arg to confirm that the words `data` and `input` are included. (loose enough to let us modify the text, but strict enough to make sure we are clear enough about what happened)